sklearn/metrics/pairwise.py

```diff
@@ -185,12 +203,16 @@ def check_pairwise_arrays(
                 "(n_queries, n_indexed). Got (%d, %d) "
                 "for %d indexed." % (X.shape[0], X.shape[1], Y.shape[0])
             )
-    elif X.shape[1] != Y.shape[1]:
+    elif not only_check_num_samples and X.shape[1] != Y.shape[1]:
         raise ValueError(
             "Incompatible dimension for X and Y matrices: "
             "X.shape[1] == %d while Y.shape[1] == %d" % (X.shape[1], Y.shape[1])
         )
-
+    elif only_check_num_samples and len(X) != len(Y):
```

Comment:
why not call `check_consistent_length`?

Reply:
The length of the input is checked in any case. We could call it `only_check_consistent_length` or `only_check_input_length_consistency` or similar if you prefer.